packages/neos-ui/src/Containers/Modals/AddNode/index.js

```diff
@@ -132,183 +127,64 @@ export default class AddNodeModal extends PureComponent {
             mode: '',
             step: 1,
             selectedNodeType: null,
-            elementValues: {}
+            elementValues: {},
+            validationErrors: null
         };
 
         this.handleSelectNodeType = this.handleSelectNodeType.bind(this);
         this.handleModeChange = this.handleModeChange.bind(this);
         this.handleDialogEditorValueChange = this.handleDialogEditorValueChange.bind(this);
         this.handleSave = this.handleSave.bind(this);
         this.handleBack = this.handleBack.bind(this);
+        this.getValidationErrors = this.getValidationErrors.bind(this);
     }
 
     render() {
-        if (!this.props.referenceNode) {
+        const {referenceNode, handleClose, nodeTypesRegistry, getAllowedNodeTypesByModeGenerator} = this.props;
+        if (!referenceNode) {
             return null;
         }
 
         if (this.state.step === 1) {
-            return this.renderStep1();
+            return (
+                <Step1
```

Comment:
I thought about it, but since here we have only to steps and we don't plan to expand this logic in the future, I left it the way @skurfuerst coded it. If you think it's really important, please suggest better names.